### PR TITLE
fix: validate the ip before attempting to  convert it

### PR DESCRIPTION
**Pull request by live627**
https://www.simplemachines.org/community/index.php?topic=576679.0

### CODE REVIEW

**Comment by albertlast** [comment]
When i read the code correct, you keep malformed ip addresses in the db instead of deleting them?

**Comment by sbulen** [comment]
~~I think albertlast is right here.  We need something to catch & deal with the invalid values.  We have seen plenty of them...~~

~~So...~~
~~- I think the invalid values need to be dealt with - null?~~
~~- I think we should accept valid ipv6 as well.  No need to throw them out.  I think lots of folks are running SleePy's ipv6 mod.~~

Works fine.  See note below.

**Review by w3bdesign — APPROVED** [review]

File: other/upgrade.php [hunk 1]
```diff
@@ -5098,7 +5098,7 @@ function MySQLConvertOldIp($targetTable, $oldCol, $newCol, $limit = 50000, $setS
 		{
 			$arIp[$i] = trim($arIp[$i]);
 
-			if (empty($arIp[$i]))
+			if (!filter_var($arIp[$i], FILTER_VALIDATE_IP, FILTER_FLAG_IPV4))
```
Comment (sbulen) [inline other/upgrade.php]: I think we should allow for IPV4 & IPV6 here

**Review by sbulen — CHANGES_REQUESTED** [review]
I think we will have a lot of folks converting from ipv6, so we should allow for ipv4 & ipv6.

Tests fine otherwise - properly filters out any non-IP.

**Comment by sbulen** [comment]
@live627 - your thoughts on adding ipv6 here?

File: other/upgrade.php (hunk 1 above)
Comment (live627) [reply other/upgrade.php]:
```suggestion
			if (!filter_var($arIp[$i], FILTER_VALIDATE_IP, FILTER_FLAG_IPV4 | FILTER_FLAG_IPV6))
```

Comment (live627) [reply other/upgrade.php]: why the newline giithub why u ignore our conffig

**Review by sbulen — APPROVED** [review]

**Review by w3bdesign — APPROVED** [review]

**Pull request merged**